### ISSUE TITLE
Limit pagination to avoid Result window is too large exceptions

### ISSUE THREAD

**Issue by Rhubbikull** [searchkit/searchkit#174]
While testing pagination, we found that SearchKit doesn't really handle the following exception particularly well

```
Caused by: QueryPhaseExecutionException[Result window is too large, from + size must be less than or equal to: [10000] but was [12500]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level parameter.]
```

As discussed @GregoryPotdevin in the live chat, it would be nicer to set a limit on the pagination so that it does not crash, and implement a `<ResultWindowIsTooLarge />` component similar to a `<NoHits />` component.

**Comment by GregoryPotdevin** [comment]
Thanks for the suggestion. Features are frozen for v0.8 so you'll have to wait a little more.

**Comment by joemcelroy** [comment]
Thanks, il assign this to 0.9 to look at options on limiting pagination. Currently NoHits component renders on an ES exception in the latest release (was implemented in 0.6) 

https://github.com/searchkit/searchkit/blob/develop/src/components/search/hits/src/NoHits.tsx#L94
